src/Orckestra.Composer.Cart/Factory/Order/OrderDetailsViewModelFactory.cs

```diff
@@ -138,10 +138,23 @@ protected virtual OrderDetailInfoViewModel GetOrderInfosViewModel(CreateOrderDet
             orderInfos.OrderStatusRaw = param.Order.OrderStatus;
             orderInfos.BillingCurrency = param.Order.Cart.BillingCurrency;
             orderInfos.PricePaid = LocalizationProvider.FormatPrice((decimal)param.Order.Cart.Total, CurrencyProvider.GetCurrency());
+            orderInfos.IsOrderEditable = IsOrderEditable(param);
 
             return orderInfos;
         }
 
+        protected virtual bool IsOrderEditable(CreateOrderDetailViewModelParam param)
```

Comment:
please add Tests for this method and check next test cases:

1) 
EditableShipmentStates = "New|Pending"
param.Order.Cart.GetAllShipmentStatuses() = "[New]"
Should be TRUE

2) 
EditableShipmentStates = "New|Pending"
param.Order.Cart.GetAllShipmentStatuses() = "[Pending]"
Should be TRUE

3) 
EditableShipmentStates = "New|Pending"
param.Order.Cart.GetAllShipmentStatuses() = "[Pending, New]"
Should be TRUE

4) 
EditableShipmentStates = "New|Pending"
param.Order.Cart.GetAllShipmentStatuses() = "[Canceled]"
Should be FALSE

5) 
EditableShipmentStates = "New|Pending"
param.Order.Cart.GetAllShipmentStatuses() = "[New,Canceled]"
Should be FALSE